### PR TITLE
NPE in BlockStructureBraces

### LINKED ISSUE #481: NPE in BlockStructureBraces when there are two nested ifs without braces
## Describe the bug
```kotlin
class Example {
    fun foo() {
        if (condition1)
            if (condition2)
                bar()
    }
}
```

## Expected behavior
All `if`s should get surrounded with braces.

## Observed behavior
```
java.lang.NullPointerException
    at org.jetbrains.kotlin.com.intellij.psi.impl.source.codeStyle.CodeEditUtil.saveWhitespacesInfo (CodeEditUtil.java:122)
    at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.ChangeUtil.saveIndentationToCopy (ChangeUtil.java:106)
    at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.ChangeUtil.copyLeafWithText (ChangeUtil.java:82)
    at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafElement.replaceWithText (LeafElement.java:154)
    at org.cqfn.diktat.ruleset.rules.BlockStructureBraces$checkOpenBraceOnSameLine$1.invoke (BlockStructureBraces.kt:181)
    at org.cqfn.diktat.ruleset.rules.BlockStructureBraces$checkOpenBraceOnSameLine$1.invoke (BlockStructureBraces.kt:53)
```

## Environment information
* diktat version: 0.1.3
* build tool (maven/gradle): maven
* how is diktat run (CLI, plugin, etc.): plugin
* kotlin version: 1.4.10

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
   Fixed bug
   closes #481

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt [hunk 1]
```diff
@@ -178,7 +178,9 @@ class BlockStructureBraces(private val configRules: List<RulesConfig>) : Rule("b
                 if (braceSpace == null || braceSpace.elementType != WHITE_SPACE) {
                     node.addChild(PsiWhiteSpaceImpl(" "), nodeBefore)
                 } else {
-                    (braceSpace as LeafPsiElement).replaceWithText(" ")
+                    val spaceParent = braceSpace.treeParent
+                    spaceParent.addChild(PsiWhiteSpaceImpl(" "), braceSpace)
+                    spaceParent.removeChild(braceSpace)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:183]: Why does this fix an error? I think `replaceWithText` does more or less the same?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt [hunk 2]
```diff
@@ -178,7 +178,7 @@ class BlockStructureBraces(private val configRules: List<RulesConfig>) : Rule("b
                 if (braceSpace == null || braceSpace.elementType != WHITE_SPACE) {
                     node.addChild(PsiWhiteSpaceImpl(" "), nodeBefore)
                 } else {
-                    (braceSpace as LeafPsiElement).replaceWithText(" ")
+                    braceSpace.treeParent.replaceWhiteSpaceText(braceSpace, " ")
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:181]: here can be problems if  treeParent is not a WhiteSpace or does not exist, need to be careful with that

Comment (kentr0w) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:181]: But above, there is a check for this case `checkBraceNode`

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Pull request merged**